cmake/BLTThirdPartyConfigFlags.cmake.in

```diff
@@ -1,18 +1,46 @@
 if ("@ENABLE_CUDA@")
-    set(BLT_ENABLE_CUDA     "@ENABLE_CUDA@")
+    set(BLT_ENABLE_CUDA           "@ENABLE_CUDA@")
 endif()
 
 if ("@ENABLE_HIP@")
-    set(BLT_ENABLE_HIP      "@ENABLE_HIP@")
+    set(BLT_ENABLE_HIP            "@ENABLE_HIP@")
 endif()
 
 if ("@ENABLE_MPI@")
-    set(BLT_ENABLE_MPI      "@ENABLE_MPI@")
+    set(BLT_ENABLE_MPI            "@ENABLE_MPI@")
 endif()
 
 if ("@ENABLE_OPENMP@")
-    set(BLT_ENABLE_OPENMP   "@ENABLE_OPENMP@")
+    set(BLT_ENABLE_OPENMP         "@ENABLE_OPENMP@")
 endif()
 
-set(BLT_ENABLE_FIND_MPI     "@ENABLE_FIND_MPI@")
-set(BLT_ENABLE_CLANG_CUDA   "@ENABLE_CLANG_CUDA")
+set(BLT_ENABLE_FIND_MPI           "@ENABLE_FIND_MPI@")
+
+# User configurable CUDA options needed by downstream.  Normally, exporting
+# these targets would pass these flags to downstream targets, but this
+# must be done manually if targets aren't exported.
+if ("@ENABLE_CLANG_CUDA@" AND NOT DEFINED ENABLE_CLANG_CUDA)
```

Comment:
+1

It took me a second to wrap my brain around this if/elseif pattern. Can you put a note to explain it?